File: Leetcode_重刷紀錄/0020.Problems2215. Find the Difference of Two Arrays/Find the Difference of Two Arrays.cpp
#include <vector>
#include <unordered_set>
using namespace std;

class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        // 使用unordered_set分別存儲兩個向量的元素，以去除重複元素並提高查找效率
        unordered_set<int> set1(nums1.begin(), nums1.end()), set2(nums2.begin(), nums2.end());

        // 分別為nums1和nums2初始化結果向量
        vector<vector<int>> ans(2);
        
        // 查找在nums1中但不在nums2中的元素
        for (int num : set1) {
            if (!set2.count(num)) {
                ans[0].push_back(num);
            }
        }
        
        // 查找在nums2中但不在nums1中的元素
        for (int num : set2) {
            if (!set1.count(num)) {
                ans[1].push_back(num);
            }
        }
        
        return ans;
    }
};

File: Leetcode_重刷紀錄/0020.Problems2215. Find the Difference of Two Arrays/Find the Difference of Two Arrays.py
class Solution:
     def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
         # 將清單轉換為集合移除重複元素，並且方便後續操作
         set1, set2 = set(nums1), set(nums2)

         # 使用集合的差集運算來找到僅存在於一個集合中的元素
         unique_to_nums1 = set1 - set2
         unique_to_nums2 = set2 - set1

         # 將結果轉換為清單格式並傳回
         return [list(unique_to_nums1), list(unique_to_nums2)]